[feat/init] Add terraform project files

2ghrms opened the pull request:
## 🎯 개요
Clokey 프로젝트의 AWS 인프라를 Terraform으로 관리하기 위한 Infrastructure as Code 초기 구성을 추가했습니다.

## 주요 구성 요소
1. Bootstrap 모듈 (terraform/bootstrap/)
S3 백엔드 버킷: Terraform 상태 파일 저장소
버전 관리 활성화
AES256 서버 사이드 암호화
공개 액세스 차단
계정별 고유 버킷명 (clokey-terraform-state-{account_id})
2. 재사용 모듈 (terraform/modules/)
Network: VPC, 서브넷, 라우팅 테이블, Internet Gateway
Compute: EC2 인스턴스 구성
Database: RDS MySQL 설정
Storage: S3 버킷 구성
Security: 보안 그룹 및 NACL
3. 환경별 설정 (terraform/env/)
Dev 환경: 개발용 인프라 구성
Prod 환경: 프로덕션용 인프라 구성
각 환경별 독립적인 변수 관리
4. CI/CD 파이프라인 (.github/workflows/)
Dev 환경: main 브랜치 push 시 자동 배포
Prod 환경: GitHub Release 생성 시 자동 배포
환경별 AWS 인증 정보 분리 관리

2ghrms committed:
Add terraform project files

yongjun0511 commented on .github/workflows/ci_dev.yml:
```diff
@@ -0,0 +1,80 @@
+        run: terraform -chdir=terraform/env/dev init
+
+      - name: Terraform Format Check (전체)
+        run: terraform fmt -check -recursive
```
현재 CI가 포메팅에서 걸리고 있습니다.

항상 push 전에 format과 validation을 실행해보고 CI를 올려주시면 감사하겠습니다!

yongjun0511 commented on .gitignore:
```diff
@@ -0,0 +1,33 @@
+
+# Terraform variables files with sensitive data
+*.tfvars
+!example.tfvars
```
이러면 example을 제외한 모든 tfvars를 무시하게 됩니다!

아래 secret.tfvars를 무시하도록 되어 있으니 이 부분은 제거해도 좋을 것 같습니다.

yongjun0511 commented on README.md:
```diff
@@ -0,0 +1,231 @@
+
+#### 자동 배포
+- **Dev 브랜치**: `main` 브랜치에 push 시 Dev 환경 자동 배포
+- **Prod 환경**: GitHub Release 생성 시 Prod 환경 자동 배포
```
저희 구조는 dev 브랜츠에 push시 dev환경 배포되고, main 브랜치에 push할 경우 prod서버에 반영됩니다!

2ghrms replied:
지금 dev 브랜치와 prod 브랜치 유지중이어서 "`prod` 브랜치에 push 시 Prod 환경 자동 배포" 로 반영하겠습니다!

yongjun0511 commented on README.md:
```diff
@@ -0,0 +1,231 @@
+## 🔒 보안 고려사항
+
+### 파일 보안
+- `*.tfvars` 파일은 `.gitignore`에 포함되어 Git에 커밋되지 않습니다
```
tfvars를 제외시 이 부분도 반영 부탁드립니다!

yongjun0511 commented on terraform/bootstrap/tf_state_bucket.tf:
```diff
@@ -0,0 +1,44 @@
+
+  tags = {
+    Name        = "Terraform State Bucket"
+    Environment = "Bootstrap"
```
Env는 Bootstrap이 아닌 dev 또는 prod로 되어야 합니다 !

따라서, 두 환경에 맞게 vars를 분리하는 것은 어떤가요!

yongjun0511 commented on terraform/bootstrap/tf_state_bucket.tf:
```diff
@@ -0,0 +1,44 @@
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
```
이런 리소스 관련된 설정들을 간편하게 할 수 있도록 모듈에 추가해놓으면 좋을 것 같아요.

나중에 S3 버킷 만들때에도 다 설정할 부분이라 모듈에 정리해도 일반적으로 잘 쓸 수 있을 것 같습니다.

yongjun0511 commented on terraform/modules/compute/ec2/main.tf:
```diff
@@ -0,0 +1,11 @@
+
+  tags = {
+    Name = var.name
+  }
```
이름 이외에 다른 태그도 받을 수 있도록 열어주세요!

이건 Ec2 뿐만 아니라 모든 모듈에 적용 부탁드립니다.

yongjun0511 commented on terraform/modules/compute/ec2/main.tf:
```diff
@@ -0,0 +1,11 @@
+  instance_type          = var.instance_type
+  subnet_id              = var.subnet_id
+  vpc_security_group_ids = var.security_group_id_list
+  user_data              = file("${path.module}/userdata.sh")  # 👈 여기 추가
```
- volume size
- volume type 
- public IP 사용 여부 (예를 들어 엘라스틱 캐시등은 필요없었음)
등을 설정할 수 있도록 열어주어야 할 것 같습니다. -> 범용적인 사용을 할 수 있도록 일반적으로 설계하면 좋습니다.

또한, ssh 키를 주입할 수 있도록 열어주어야 할 것 같아요. GUI를 통해서 EC2를 만들 때는 pem key를 생성하고 동시에 Ec2생성이 가능했지만, 테라폼으로 할 때는 미리 만든 pem key 주입이 필요합니다!

이 부분은 변수만 열어주고 aws에서 수동으로 만든 이후에 이름만 넣어도 좋을 것 같아요.

yongjun0511 commented on terraform/modules/compute/ec2/main.tf:
```diff
@@ -0,0 +1,11 @@
+  subnet_id              = var.subnet_id
+  vpc_security_group_ids = var.security_group_id_list
+  user_data              = file("${path.module}/userdata.sh")  # 👈 여기 추가
+
```
추가적으로, userdata.sh가 초기 설정과 관련된 부분이 맞겠죠?

저걸 열어줄수 있다면 주입할 수 있도록 열어만 주고 모듈에서 구현을 할 필요는 없을 것 같아요 ! 
모든 ec2가 java가 깔릴 필요는 없으니까요~!

+ 저희 WAS(api용) ec2에서는 도커를 아마도 사용할 것 같아요. 
그래서 도커까지 깔아주시는 방향으로 확인 부탁드립니다.

yongjun0511 commented on terraform/modules/database/rds/main.tf:
```diff
@@ -0,0 +1,19 @@
+  db_subnet_group_name   = aws_db_subnet_group.this.name
+  vpc_security_group_ids = [var.security_group_id]
+  skip_final_snapshot    = true
+
```
engine_version          = var.engine_version
  instance_class          = var.instance_class
  publicly_accessible     = var.publicly_accessible
  backup_retention_period = var.backup_retention_period
  parameter_group_name    = aws_db_parameter_group.main.name (파라미터 그룹은 리소스 추가해주세요)
  
  등등 추가해 주시면 감사하겠습니다.

yongjun0511 commented on terraform/modules/compute/ec2/variables.tf:
```diff
@@ -0,0 +1,12 @@
+  type        = list(string)
+}
+
+variable "name"                { type = string }
```
모듈도 나중에 보고 쓰는거라 모든 입력값 출력값에 description을 간단하게라도 작성해 주시면 감사 하겠습니다.

yongjun0511 commented on terraform/modules/storage/s3/variables.tf:
```diff
@@ -0,0 +1,32 @@
+variable "enable_block_public_access" {
+  description = "Whether internet access blocking is enabled"
+  type        = bool
+  default     = true
```
현재 모듈에서 이름을 제외한 변수들이 사용되지 않고 있습니다!

위의 설정들을 main.tf에서 사용해 주세요.

yongjun0511 commented on terraform/env/dev/backend.tf:
```diff
@@ -0,0 +1,10 @@
+# S3 Backend Configuration
+# 로컬 테스트용 (실제 값 사용)
```
실제 dev 서버랑 연결된 버킷이라 주석을 없애도 좋을 것 같습니다!

yongjun0511 commented on terraform/env/dev/variables.tf:
```diff
@@ -0,0 +1,57 @@
+  description = "AWS Account ID"
+  type        = string
+  sensitive   = true
+}
```
vars로 빼주는 이유에 대해서 저는 다음과 같은 생각을 가지고 있습니다.

- 모듈에서는 입력값으로 재사용하기 위해서 입력할 수 있는 값들을 당연히 빼주어야 한다.
- 그런데, 이런 부분들을 env환경에서 뺴줄 필요는 없다고 생각했습니다.

"이미 만들어진 모듈을 적용해서 인스턴스를 만드는 과정에 있는데 또 vars로 뺴줄 필요가 있을까?"라는 생각했습니다.

그래서 저는 민감한 정보만 vars로 뺴주고 나머지는 보통 인스턴스에 하드 코딩으로 넣어주었습니다. 로컬에서 공통적으로 사용할 변수들은 어짜피 locals로 빼주면 되니까요!

- 위 과정의 또 다른 근거는 로컬에서 테라폼을 실행한다면 vars를 tfvars로 설정할 수 있지만, CI/CD 파이프라인을 구축하게 되면 이 vars를 모두 깃허브 시크릿에 등록해주고 CI/CD에서 주입을 해주어야 합니다.

그래서 제가 설계했던 CI/CD는 4개의 값만 주입이 되고 관리되는 대상은 다음과 같았습니다.

```
variable "access_key_id" {
  description = "AWS access key id"
  type        = string
  sensitive   = true
}

variable "secret_access_key" {
  description = "AWS secret access key"
  type        = string
  sensitive   = true
}

variable "username" {
  description = "RDS 마스터 사용자 이름"
  type        = string
}

variable "password" {
  description = "RDS 마스터 사용자 비밀번호"
  type        = string
  sensitive   = true
}

```

혹시나 생각하시는 다른 방법이 있다면 꼭 알려주세요 ! 
그런데 우선 CI/CD를 통해서 인스턴스 생성을 자동화 하기 위해서는 개편이 필요해 보입니다.

yongjun0511 commented on terraform/env/dev/provider.tf:
```diff
@@ -0,0 +1,7 @@
+  default_tags {
+    tags = local.common_tags
+  }
+}
```
env가 아니라 CI/CD 파이프라인을 통한 생성을 위해서는 아마 여기서 AWS관련 Secret들 주입이 필요할겁니다.

yongjun0511 commented on terraform/env/dev/outputs.tf:
```diff
@@ -0,0 +1,45 @@
+# EC2 Outputs
```
Outputs는 이미 모듈에서 만들어놔서 여기서 또 선언할 필요는 없습니다!

그냥 필요한 곳에서 주입해서 사용하기만 하면 됩니다!

yongjun0511 commented on terraform/env/dev/network.tf:
```diff
@@ -0,0 +1,145 @@
+  source = "../../modules/network/igw"
+  vpc_id = module.vpc.vpc_id
+  name   = "${local.name_prefix}-igw"
+}
```
일반적인 Cloud환경을 생각한다면, 보통 dev와 prod EC2이런게 공존하고 환경별 인스턴스가 하나의 계정에서 공존하고 있어요.
(물론 저희가 적용할 구조는 가격을 절감하기 위해 분리할 예정이긴 합니다.)

그래서, 공유되는 리소스에는 environment를 붙이지 않아요.
대표적인 예시가 vpc와 igw같이 분리할 필요가 없는 부분들 입니다! 

따라서, 이런 부분들은 사실 모듈에서 네이밍 조건을 걸어두는게 좋다고 생각합니다. 
하지만 시간적인 부분을 생각해서 참고만 하시면 좋을 것 같아요 ! 시간이 남는다면 이 부분은 모듈을 개편해 주셔도 좋습니다.

yongjun0511 commented on terraform/env/dev/compute.tf:
```diff
@@ -0,0 +1,10 @@
+  ami             = data.aws_ami.amazon_linux_2023.id
+  instance_type   = "t3.micro"
+  subnet_id       = module.subnet_public_a.subnet_id
+  name            = "${local.name_prefix}-ec2"
```
API 서버로 쓰일 것 같으니 맨 뒤에 -api가 좋을 것 같슴다!

yongjun0511 commented on terraform/env/dev/locals.tf:
```diff
@@ -0,0 +1,29 @@
+  private_subnets = {
+    a = "10.0.11.0/24"
+    c = "10.0.12.0/24"
+  }
```
가용영역, 서브넷 이런 부분들도 로컬에서 공유되는 변수가 아니기 때문에 그냥 subnet 선언과 vpc선언부에서 하드 코딩해도 괜찮을 것 같아요!

yongjun0511 commented on terraform/env/dev/database.tf:
```diff
@@ -0,0 +1,16 @@
+  username           = var.rds_username
+  security_group_id  = module.sg.security_group_id
+}
+
```
Security Group은 EC2와 RDS는 다른 것을 사용해야 합니다 ! 

저희는 MySQL을 사용하니 3306이 열려있는 것으로요 ! 
+ RDS 서브넷 그룹고 파라미터 그룹 적용 부탁드립니다.

2ghrms committed:
feat/init: PR 코멘트 반영

- terraform fmt -recursive & terraform validate 실행

2ghrms committed:
feat/init: PR 코멘트 반영 - README.md 수정

- 배포 관련 브랜치 설명 코드와 일치화
- secret.tfvars 이외에 제외

2ghrms committed:
feat/init: PR 코멘트 반영 - gitignore 수정

- tfvars 관련 ignore 파일명 수정

2ghrms committed:
feat/init: PR 코멘트 반영 - bootstrap 관련 수정

- S3 모듈 추가 정의
- vars에 environment  추가
- S3 모듈 사용법 README 추가

2ghrms committed:
feat/init: PR 코멘트 반영 - gitignore 수정

- tfvars 관련 ignore 파일명 수정 후 terraform.tfvars 반영됨

2ghrms committed:
feat/init: PR 코멘트 반영 - gitignore 수정

- tfvars 관련 ignore 파일명 수정 후 terraform.tfvars 반영됨

2ghrms committed:
feat/init: PR 코멘트 반영 - tag 통일 및 추가 열어둠

- 각 모듈 variable.tf에 purpose와 environment, tags 추가
- 각 모듈 main.tf에 tags 정의
- compute, database 등에 적용

2ghrms committed:
feat/init: PR 코멘트 반영 - 주석 제외

2ghrms committed:
feat/init: PR 코멘트 반영 - EC2 개편

- volume size 입력받음
- volume type 입력받음
- public IP 사용 여부 입력받음
- SSH 키 주입
- was-userdata.sh로 wsa용 userdata 예시 추가

2ghrms committed:
feat/init: PR 코멘트 반영 - EC2 개편

- volume size 입력받음
- volume type 입력받음
- public IP 사용 여부 입력받음
- SSH 키 주입
- was-userdata.sh로 wsa용 userdata 예시 추가

2ghrms committed:
feat/init: PR 코멘트 반영 - RDS 개편

- engine_version, instance_class, publicly_accessible, backup_retention_period, parameter_group_name 추가

2ghrms committed:
feat/init: PR 코멘트 반영 - description 추가

- variables.tf에 description 추가
- output.tf에 description 추가

2ghrms committed:
feat/init: PR 코멘트 반영 - 모든 variables 사용하도록 모듈 수정

2ghrms committed:
feat/init: PR 코멘트 반영 - env에서 vars로 빼지 않고 locals.tf로 적용하도록 수정

- dev 적용
- prod 적용

2ghrms committed:
feat/init: PR 코멘트 반영 - env에서 vars로 빼지 않고 locals.tf로 적용하도록 수정

- dev 적용
- prod 적용
- terraform format & validate

2ghrms committed:
feat/init: PR 코멘트 반영 - env의 outputs.tf 삭제

- dev 적용
- prod 적용

2ghrms committed:
feat/init: PR 코멘트 반영 - 태그 및 이름 수정

- VPC, IGW, Route Tables -> 공유 리소스에 environment 태그 삭제
- compute.tf -> api 서버 명시

2ghrms committed:
feat/init: PR 코멘트 반영 - locals.tf 수정

- 가용영역, 서브넷 부분 하드코딩으로 수정

2ghrms committed:
feat/init: PR 코멘트 반영 - EC2와 RDS 환경 분리

- EC2와 RDS의 보안그룹 분리
- RDS 보안그룹에 3306 포트 열어둠
- 파라미터 그룹 적용

2ghrms committed:
feat/init: PR 코멘트 반영 - fomat 검사

2ghrms committed:
feat/init: PR 코멘트 반영 - cicd vars 추가

2ghrms committed:
feat/init: domain dev, prod 분기

2ghrms committed:
feature/init: ACM과 ALB 모듈 추가 및 CI 최적화

- ACM 모듈 추가 (SSL 인증서 관리)
- ALB 모듈 추가 (Application Load Balancer)
- dev/prod 환경에 ACM과 ALB 적용
- CI 워크플로우 최적화 (terraform plan 변수 최소화)
- DB ID 통일 ('clokey_db'로 설정)
- Route53 모듈 개선 (ALB alias 레코드 지원)
- 보안 강화 (EC2는 ALB를 통해서만 접근 가능)

2ghrms committed:
Merge remote-tracking branch 'origin/feature/init'

2ghrms committed:
feat/init: ci 수정

2ghrms committed:
fix: Terraform CI 오류 및 Route53 count 의존성 문제 해결

- CI에서 user_data 변수를 tfvars 파일로 분리하여 'Too many command line arguments' 오류 해결
- Route53 모듈의 count 조건을 개선하여 ALB 의존성 문제 해결
- Route53 hosted zone과 record 생성을 분리하여 의존성 명확화
- terraform fmt로 코드 포맷팅 정리

Changes:
- .github/workflows/ci_dev.yml: user_data를 별도 파일로 분리
- terraform/env/dev/network.tf: Route53 모듈 분리 및 의존성 추가
- terraform/modules/network/route53/main.tf: count 조건 개선
- terraform/modules/network/route53/output.tf: 안전한 리소스 접근으로 수정

2ghrms committed:
fix: Route53 모듈 record_name 변수 필수 오류 해결

- record_name 변수에 default = null 추가
- route53_zone 모듈에서 레코드 생성하지 않을 때 record_name 불필요
- 'Missing required argument' 오류 해결

2ghrms committed:
fix: Route53 record_name null 체크 추가

2ghrms committed:
fix: Route53 count 의존성 문제 해결

yongjun0511 commented on README.md:
````diff
@@ -0,0 +1,230 @@
+terraform init
+terraform plan -var-file="terraform.tfvars"
+terraform apply -var-file="terraform.tfvars"
+```
````
환경 설정 부분도 자동화가 된거죠!!?

yongjun0511 commented on terraform/env/dev/data.tf:
```diff
@@ -0,0 +1,26 @@
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+}
```
요부분 가장 최근 AMI로 불러올 수 있습니다 (실행 기준).

```
data "aws_ami" "ubuntu_latest" {
  most_recent = true

  filter {
    name   = "name"
    values = ["ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-*"]
  }

  filter {
    name   = "virtualization-type"
    values = ["hvm"]
  }

  owners = ["099720109477"]
}
```

요걸로 반영 어떠신가요

yongjun0511 commented on terraform/env/dev/network.tf:
```diff
@@ -0,0 +1,274 @@
+  subnet_ids      = [module.subnet_public_a.subnet_id, module.subnet_public_c.subnet_id]
+  vpc_id          = module.vpc.vpc_id
+
+  target_group_port     = 8080
```
현재 배포 구조를 그대로 사용하겠습니다 ! 80으로 요청 넣어주세요!

yongjun0511 commented on terraform/env/dev/network.tf:
```diff
@@ -0,0 +1,274 @@
+      use_cidr    = true
+      use_sg      = false
+      cidr_blocks = ["0.0.0.0/0"]
+    },
```
일단 443이랑 8080은 제외하져! 

ALB로 80으로 넣어주게 사용합시당

yongjun0511 commented on terraform/env/dev/storage.tf:
```diff
@@ -0,0 +1,8 @@
+  environment = local.environment
+  purpose     = "storage"
+}
+
```
public 버킷으로 사용하되 외부 요청은 Get만 허용하도록 하죠.

AWS sdk를 통해서 업로드랑 삭제를 처리를 하는 구조입니다!

정책은 json으로 설정하는데,

```
resource "aws_s3_bucket_policy" "public_read" {
  bucket = aws_s3_bucket.example.id

  policy = jsonencode({
    Version = "2012-10-17"
    Statement = [
      {
        Effect    = "Allow"
        Principal = "*"
        Action    = [
          "s3:GetObject"
        ]
        Resource = "${aws_s3_bucket.example.arn}/*"
      }
    ]
  })
}
```

아마 이런 느낌일텐데 확인 한 번 부탁드릴게요~

yongjun0511 commented on terraform/env/dev/terraform.tfvars:
```diff
@@ -0,0 +1,14 @@
+rds_username = "admin"
+
+# Route53 설정 (기본값)
+domain_name = "dev.clokey.store"
```
도메인도 공개하지 않는 것이 좋아보입니다.

yongjun0511 commented on terraform/env/dev/compute.tf:
```diff
@@ -0,0 +1,36 @@
+# EC2 Instance
+module "ec2" {
+  source                 = "../../modules/compute/ec2"
+  ami                    = data.aws_ami.amazon_linux_2023.id
```
이 부분에서 최신 이미지 가져와 주세요~

yongjun0511 commented on terraform/env/dev/compute.tf:
```diff
@@ -0,0 +1,36 @@
+  associate_public_ip_address = true
+
+  # 루트 볼륨 설정
+  root_volume_size      = 20
```
루트 볼륨 prod와 동일하게 30으로 하시죠!

yongjun0511 reviewed (commented):
댓글 남긴 부분 수정 후 테스트만 진행해보면 될 것 같아요.

(현재 사용하는 AWS 서버에서 모든 것을 terminate 후 생성 해보기)

이 과정에서, 새로운 AWS계정을 만들고 MFA 인증까지 설정한 뒤... 수동적으로 해야하는 부분들을 모두 기록해 주시면 됩니다. (처음 보는 사람도 잘 따라할 수 있도록) 이 과정에서 사진도 첨부해주시면 좋을 것 같아요.

사진은 로컬 보다는 PR 올려놓고 편집해서 드래그 드롭하면 들어갑니다!

AWS 정책 변화에 따라서 6개월 마다 이사를 다녀야 하기 때문에.. 꼼꼼히 정리 부탁드릴게요!

2ghrms committed:
feature/init: PR 피드백 반영한 개선사항

- Ubuntu 22.04 LTS AMI로 최신 이미지 사용
- ALB 타겟 그룹을 8080에서 80 포트로 변경
- 보안 그룹 최적화: 443/8080 제거, 80/22만 유지
- S3 public read 정책 추가 (GET만 허용)
- 민감한 정보(도메인, RDS 자격증명)를 secret 파일로 분리
- Dev 환경 루트 볼륨 크기를 30GB로 증가 (Prod와 동일)
- GitHub Secrets와 환경 설정 자동화 완성

2ghrms committed:
feat/init: CI 파이프라인에서 RDS 자격증명 통합

2ghrms committed:
feat/init: db storage 30으로 증가

2ghrms committed:
feat/init: route53 설정 통일 -> hosted zone 생성

2ghrms committed:
feat/init: userdata.sh 수정

- swap 메모리 2GB 설정
- redis container 설정

2ghrms committed:
feat/init: too many comands line 해결 위해 secrets를 terraform.tfvars로 처리

2ghrms committed:
feat/init: 버킷 통일

2ghrms committed:
feat/init: S3 백엔드 적용

2ghrms committed:
feat/init: CI에서 userdata 사용하지 않도록 수정

2ghrms committed:
feat/init: CD 조건 변경 (이후에 롤백 예정)

2ghrms committed:
feat/init: userdata.sh 파일 읽도록 수정

2ghrms committed:
feat/init: cd 에러 해결

- EC2 KeyPair 이름 문제 해결
- MySQL 버전을 8.0.35 → 8.0.34로 변경
- 인증서 검증 타임아웃을 10분 → 20분으로 증가
- Route53 레코드 의존성 명시적 추가
- S3 버킷 중복 문제
- website_endpoint → website_domain으로 변경

2ghrms committed:
feature/init: Terraform apply 오류들 해결

- EC2 KeyPair, RDS MySQL 버전, S3 버킷 충돌 문제 해결
- ACM 인증서 검증 및 UserData 로깅 개선
- deprecated 속성 수정

2ghrms committed:
feature/init: ci fmt문제 해결

2ghrms committed:
feature/init: cd fmt문제 해결

- ALB 설정

2ghrms committed:
feat/init: nginx 포함 userdata.sh 적용

2ghrms committed:
feature/init: fmt 에러 해결

2ghrms committed:
feat/init: rds password 추가

2ghrms committed:
feat/init: vpc environment 추가

2ghrms committed:
feat/init: rds password 추가

2ghrms committed:
feat/init: 의도한 이름이 보이도록 수정

2ghrms committed:
feat/init: userdata base64 인코딩

2ghrms committed:
feat/init: rds와 s3 이름 변경

2ghrms committed:
feat/init: s3 버킷명 충돌로 이름 수정

2ghrms committed:
feat/init: rds 초기화

2ghrms committed:
feat/init: rds 초기화 롤백

2ghrms committed:
feat/init: mysql 설치 추가 및 userdata.sh 경로 수정

2ghrms committed:
feat/init: cicd에서 userdata.sh 제외

2ghrms committed:
feat/init: userdata.sh base64 encoding

2ghrms committed:
feat/init: userdata.sh local.tf encoding

2ghrms committed:
feat/init: cd rollback, not replace ec2 on change userdata
